selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/ImportWizardFormTest.java

```diff
@@ -0,0 +1,614 @@
+    projectExplorer.waitItem(currentProjectName);
+    projectExplorer.waitAndSelectItemByName(currentProjectName);
+    loader.waitOnClosed();
+    menu.runCommand(GIT, BRANCHES);
+    git.waitBranchInTheListWithCoState(SECOND_BRANCH);
+    git.closeBranchesForm();
+
+    projectExplorer.waitItem(currentProjectName);
+    projectExplorer.expandPathInProjectExplorerAndOpenFile(
+        currentProjectName + "/src/main/java/org.eclipse.qa.examples", "AppController.java");
+    editor.waitActive();
+  }
+
+  @Test(priority = 6)
+  public void checkImportProjectInBranchFromGitHub() throws Exception {
+    currentProjectName = importBranchRepo.getName();
+    projectExplorer.waitProjectExplorer();
+
+    importIntoBranchFromGitHub();
+    projectWizard.waitCreateProjectWizardForm();
+    projectWizard.selectTypeProject(Wizard.TypeProject.MAVEN);
+    loader.waitOnClosed();
+    projectWizard.clickSaveButton();
+    loader.waitOnClosed();
+    projectWizard.waitCreateProjectWizardFormIsClosed();
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(currentProjectName);
+    projectExplorer.waitAndSelectItemByName(currentProjectName);
+    loader.waitOnClosed();
+    menu.runCommand(TestMenuCommandsConstants.Git.GIT, TestMenuCommandsConstants.Git.BRANCHES);
+    git.waitBranchInTheListWithCoState(THIRD_BRANCH);
+    git.closeBranchesForm();
+    projectExplorer.openItemByPath(currentProjectName);
+    loader.waitOnClosed();
+    projectExplorer.openItemByPath(currentProjectName + "/AppController.java");
+    editor.waitActive();
+  }
+
+  @Test(priority = 7)
+  public void checkImportProjectSubmoduleByHttpsUrl() throws Exception {
+    currentProjectName = submoduleRepo.getName() + "Https";
+    projectExplorer.waitProjectExplorer();
+
+    importRecursivelyFromGitUrl(submoduleRepo.getHtmlUrl(), currentProjectName);
+    openSubmoduleOne(currentProjectName);
+    openSubmoduleTwo(currentProjectName);
+  }
+
+  @Test(priority = 8)
+  public void checkImportProjectSubmoduleBySshUrl() throws Exception {
+    currentProjectName = submoduleRepo.getName() + "Ssh";
+    projectExplorer.waitProjectExplorer();
+
+    importRecursivelyFromGitUrl(submoduleRepo.getSshUrl(), currentProjectName);
+    openSubmoduleOne(currentProjectName);
+    openSubmoduleTwo(currentProjectName);
+  }
+
+  @Test(priority = 9)
+  public void checkImportProjectSubmoduleFromGithub() throws Exception {
+    currentProjectName = submoduleRepo.getName();
+    projectExplorer.waitProjectExplorer();
+
+    importRecursivelyFromGitHub(currentProjectName);
+    openSubmoduleOne(currentProjectName);
+    openSubmoduleTwo(currentProjectName);
+  }
+
+  private void importRecursivelyFromGitUrl(String url, String projectName) throws Exception {
+    loader.waitOnClosed();
+    menu.runCommand(
+        TestMenuCommandsConstants.Workspace.WORKSPACE,
+        TestMenuCommandsConstants.Workspace.IMPORT_PROJECT);
+    importProject.waitMainForm();
```

Comment:
It's hard to understand test scenario when we have just list of tens of commands.
Please divide separate steps with empty rows and comments.